### PR TITLE
Added Solution for Cofense Intelligence.

### PR DESCRIPTION
**Pull request by jayeshprajapaticrest**
Required items, please complete
   
   Change(s):
   - Added Cofense Intelligence Solution for certification.

   Reason for Change(s):
   - New product

   Version Updated:
   - Initial Version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Workbooks/Images/Preview/CofenseThreatIntelligenceWhite1.png
Comment (v-prasadboke) [inline Workbooks/Images/Preview/CofenseThreatIntelligenceWhite1.png:1]: Please correct the name of this png file, rename it to CofenseThreatIntelligenceBlack1 as it has black background

File: Workbooks/Images/Preview/CofenseThreatIntelligenceBlack1.png
Comment (v-prasadboke) [inline Workbooks/Images/Preview/CofenseThreatIntelligenceBlack1.png:1]: Please correct the name of this png file, rename it to CofenseThreatIntelligenceWhite1 as it has white background

File: Workbooks/WorkbooksMetadata.json [hunk 1]
```diff
@@ -3682,5 +3682,26 @@
...
+    "dataTypesDependencies": [
+        "ThreatIntelligenceIndicator",
+        "Malware_Data"
+    ],
+    "dataConnectorsDependencies": [
+        "CofenseIntelligenceDataConnector"
+    ],
+    "previewImagesFileNames": [
+        "CofenseIntelligenceWhite1.png",
+        "CofenseIntelligenceBlack1.png"
```
Comment (v-prasadboke) [inline Workbooks/WorkbooksMetadata.json:3699]: Please correct the values used in "previewImagesFileNames" parameter.
Values should be :- CofenseThreatIntelligenceBlack1.png
                               CofenseThreatIntelligenceWhite1.png
Which will redirect to the images name

File: .script/tests/KqlvalidationsTests/CustomTables/Malware_data_CL.json
Comment (v-prasadboke) [inline .script/tests/KqlvalidationsTests/CustomTables/Malware_data_CL.json:1]: Please replace the d with D in the file name . It should be **Malware_Data_CL**

File: Sample Data/Custom/CofenseIntelligence/Malware_data_CL.csv
Comment (v-prasadboke) [inline Sample Data/Custom/CofenseIntelligence/Malware_data_CL.csv:1]: Same is the case here too with  d & D in the file name . It should be Malware_Data_CL

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest please work on the changes requested above

File: Workbooks/Images/Preview/CofenseThreatIntelligenceWhite1.png
Comment (jayeshprajapaticrest) [reply Workbooks/Images/Preview/CofenseThreatIntelligenceWhite1.png:1]: Done

File: Workbooks/Images/Preview/CofenseThreatIntelligenceBlack1.png
Comment (jayeshprajapaticrest) [reply Workbooks/Images/Preview/CofenseThreatIntelligenceBlack1.png:1]: Corrected as expected. Done

File: Workbooks/WorkbooksMetadata.json (hunk 1 above)
Comment (jayeshprajapaticrest) [reply Workbooks/WorkbooksMetadata.json:3699]: Corrected as expected. Done

File: .script/tests/KqlvalidationsTests/CustomTables/Malware_data_CL.json
Comment (jayeshprajapaticrest) [reply .script/tests/KqlvalidationsTests/CustomTables/Malware_data_CL.json:1]: Done

File: Sample Data/Custom/CofenseIntelligence/Malware_data_CL.csv
Comment (jayeshprajapaticrest) [reply Sample Data/Custom/CofenseIntelligence/Malware_data_CL.csv:1]: Done

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest please work on the changes requested above

@v-prasadboke Changes are done.

**Comment by v-prasadboke** [comment]
Ok @jayeshprajapaticrest will give you an update till 7-07

**Comment by jayeshprajapaticrest** [comment]
> Ok @jayeshprajapaticrest will give you an update till 7-07

OK Understood. Thanks

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest as per latest guidelines issued, solution should be repackaged with V3 tool 
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md
you can go through this readme file for the instructuons

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest please share the release notes as well

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest please go through this readme file to know more about this
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest we are waiting for your response

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest we are waiting for your response
@v-prasadboke 
Yes working on it. Getting some validation error after generating solution with V3 tool.

**Comment by jayeshprajapaticrest** [comment]
@v-prasadboke 
I have added ReleaseNotes.md and Updated PR with the solution package created using V3 tool.
So can you please proceed with the Merging process if everything looks good?

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest sorry for the inconvenience, but we are still investigating this PR.
You can expect an update till 25 Jul 2023

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest please update the sample data and custom table of **ThreatIntelligenceIndicator.json**
As threat_id is missing from these both.

As this solution contains only 2 content which is workbook and Data connector, both of them depends on the query which uses schemas and data from the sample data ingested.
Due to this im unable to test both the content, hence i request you to update the sample data and custom table.

image

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, can you provide any updates for above comments. Thanks, Prasad Boke.

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest, can you provide any updates for above comments. Thanks, Prasad Boke.

@v-prasadboke "threat_id" is not actual field of any custom table. It gets extracted from "Tags" field using parse in the query automatically. So need to add it in the sample data. "Tags" is already available.

We can connect if you want to need more details on it if required.
Thanks.

**Comment by v-prasadboke** [comment]
Thank you @jayeshprajapaticrest for the updates, will schedule a call for this and CODE QL exceptions for better clarity.

Are you available for 2:30 PM (IST) tomorrow 26 Jul 2023.

**Comment by jayeshprajapaticrest** [comment]
> Thank you @jayeshprajapaticrest for the updates, will schedule a call for this and CODE QL exceptions for better clarity.
> 
> Are you available for 2:30 PM (IST) tomorrow 26 Jul 2023.

Yes. Please share the invite.

**Comment by v-prasadboke** [comment]
Ok sure @jayeshprajapaticrest, I'll schedule a call for 2:30 PM IST next Tuesday i.e 01 Aug 2023. 
And Thank you for the updates.

**Comment by jayeshprajapaticrest** [comment]
> Ok sure @jayeshprajapaticrest, I'll schedule a call for 2:30 PM IST next Tuesday i.e 01 Aug 2023. And Thank you for the updates.

Sure. Please share the invite for the same.

**Comment by v-prasadboke** [comment]
Invite sent @jayeshprajapaticrest. Thank you.

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, If possible can you please rename the values for which the code QL validation is failing to something else. That would be very helpful. Thank you.

**Comment by v-prasadboke** [comment]
Other than that only change which is requested, Everything looks good.

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, we are waiting for the updates. Thank you.

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest, we are waiting for the updates. Thank you.

@v-prasadboke 
Can we connect to discuss about the Codeql validation error?

**Comment by jayeshprajapaticrest** [comment]
> @v-prasadboke 12:15 PM tomorrow should works for me.

@v-prasadboke 
Can you please share the link to connect at 12:15 PM today?

**Comment by v-prasadboke** [comment]
> > @v-prasadboke 12:15 PM tomorrow should works for me.
> 
> @v-prasadboke Can you please share the link to connect at 12:15 PM today?

<div style="margin-top:24px; margin-bottom:20px"><span style="font-size:24px; color:#252424">Microsoft Teams meeting</span>
</div>
<div style="margin-bottom:20px">
<div style="margin-top:0px; margin-bottom:0px; font-weight:bold"><span style="font-size:14px; color:#252424">Join on your computer, mobile app or room device</span>
</div>
<a style="font-size:14px; font-family:'Segoe UI Semibold','Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif; text-decoration:underline; color:#6264a7" class="me-email-headline" href="https://teams.microsoft.com/l/meetup-join/19%3ameeting_YzkyY2I1MjEtOGE1Yi00M2FjLWE0YzEtNDZhMGY5NjJjM2Rl%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2228263f0a-981b-4a56-a4c9-1e91e9ea44fb%22%7d">Click
 here to join the meeting</a> </div>
<div style="margin-bottom:20px; margin-top:20px">
<div style="margin-bottom:4px"><span style="font-size:14px; color:#252424" data-tid="meeting-code">Meeting ID:
<span style="font-size:16px; color:#252424">239 433 783 290</span> </span><br>
<span style="font-size:14px; color:#252424">Passcode: </span><span style="font-size:16px; color:#252424">6mVVhw
</span>
<div style="font-size:14px"><a style="font-size:14px; text-decoration:underline; color:#6264a7; font-family:'Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif" class="me-email-link" href="https://www.microsoft.com/en-us/microsoft-teams/download-app">Download
 Teams</a> | <a style="font-size:14px; text-decoration:underline; color:#6264a7; font-family:'Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif" class="me-email-link" href="https://www.microsoft.com/microsoft-teams/join-a-meeting">
Join on the web</a></div>
</div>
</div>
<div style="margin-bottom:4px">
<div style="margin-bottom:4px"><span style="font-size:14px; color:#252424; font-weight:bold">Or call in (audio only)</span>
</div>
<div style="margin-bottom:4px"><a style="font-size:14px; text-decoration:underline; color:#6264a7" href="tel:+13238494874,,687710123#">[+1 323-849-4874,,687710123#](tel:+13238494874,,687710123#)</a>
<span style="font-size:14px; color:#252424">&nbsp; United States, Los Angeles </span></div>
</div>
<span style="font-size:14px; color:#252424">Phone Conference ID: </span><span style="font-size:16px; color:#252424">687 710 123#
</span>
<div style="margin-bottom:20px"><a style="font-size:14px; text-decoration:underline; color:#6264a7; font-family:'Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif" class="me-email-link" href="https://dialin.teams.microsoft.com/8551f4c1-bea3-441a-8738-69aa517a91c5?id=687710123">Find
 a local number</a> | <a style="font-size:14px; text-decoration:underline; color:#6264a7" class="me-email-link" href="https://dialin.teams.microsoft.com/usp/pstnconferencing">
Reset PIN</a> </div>
<div style="margin-bottom:24px; margin-top:20px"><a style="font-size:14px; text-decoration:underline; color:#6264a7; font-family:'Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif" class="me-email-link" href="https://aka.ms/JoinTeamsMeeting">[Learn More](https://aka.ms/JoinTeamsMeeting)</a>
 | <a style="font-size:14px; text-decoration:underline; color:#6264a7; font-family:'Segoe UI','Helvetica Neue',Helvetica,Arial,sans-serif" class="me-email-link" href="https://teams.microsoft.com/meetingOptions/?organizerId=28263f0a-981b-4a56-a4c9-1e91e9ea44fb&amp;tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&amp;threadId=19_meeting_YzkyY2I1MjEtOGE1Yi00M2FjLWE0YzEtNDZhMGY5NjJjM2Rl@thread.v2&amp;messageId=0&amp;language=en-US">
Meeting options</a> </div>

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, as discussed in the call you can Change the values for which the Code QL check is failing.

**Comment by jayeshprajapaticrest** [comment]
@v-prasadboke 
As discussed in call I updated the constant name as well as tried by giving the value direct from the environment variable. But still I am getting the same CodeQL error.

I go through the more detail on the error and as per that we need to encrypt and decrypt the constant password values for the usage.

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, Thanks for the updates. As mentioned above, please go with it as needed.

**Comment by jayeshprajapaticrest** [comment]
@v-prasadboke Done and also Code-QL error gets resolved.

**Comment by jayeshprajapaticrest** [comment]
Can you please proceed with this and approve and merge as early as possible. Because it already took so much time in single issue?

**Comment by v-prasadboke** [comment]
Thank you @jayeshprajapaticrest for committing the change. Will have a look in it and update you about the same.

**Comment by jayeshprajapaticrest** [comment]
> Thank you @jayeshprajapaticrest for committing the change. Will have a look in it and update you about the same.

@v-prasadboke Thanks

**Comment by v-prasadboke** [comment]
Hello @jayeshprajapaticrest, I've committed some minimal changes. Please take a look and confirm once.

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest, I've committed some minimal changes. Please take a look and confirm once.
@v-prasadboke 
Thanks for the update. Will check and let you know.

**Comment by v-prasadboke** [comment]
Ok sure, Thank you @jayeshprajapaticrest.

**Comment by jayeshprajapaticrest** [comment]
> Hello @jayeshprajapaticrest, I've committed some minimal changes. Please take a look and confirm once.

@v-prasadboke Changes looks good. Can you please proceed further for merging?

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**